components/rag-wrapper/src/rag_wrapper/base/base_rag_ops.py

```
@@ -0,0 +1,115 @@
+        """Check if response indicates failure requiring retry."""
+        if hasattr(result, "response"):
+            response_text = str(result.response)
+        else:
+            response_text = str(result)
+
+        # Check for empty responses or known error patterns
+        return (
+            response_text == ""
+            or response_text == "504.0 GatewayTimeout"
+            or "timeout" in response_text.lower()
+            or len(response_text.strip()) == 0
```

Ruff (0.15.17)

[warning] Consider merging multiple comparisons: `response_text in {"", "504.0 GatewayTimeout"}`.

Merge multiple comparisons

(PLR1714)